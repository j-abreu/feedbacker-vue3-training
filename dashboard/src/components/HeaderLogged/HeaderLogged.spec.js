import { shallowMount } from "@vue/test-utils"
import HeaderLogged from '.'
import { routes } from '../../router'

import { createRouter, createWebHistory } from 'vue-router'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

const mockStore = { currentUser: {} }
jest.mock('../../hooks/useStore', () => {
  return () => {
    return mockStore
  }
})

describe('<HeaderLogged />', () => {
  it('should render header logged', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render 3 dots when there\'s not a user logged in', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router]
      }
    })

    const logoutButton = wrapper.find('#logout-button')
    expect(logoutButton.text()).toBe('...')
  })

  it('should render user name when there\'s a user logged in', async () => {
    const userName = 'Jeremias Abreu'

    router.push('/')
    await router.isReady()
    mockStore.currentUser.name = userName
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router]
      }
    })

    const logoutButton = wrapper.find('#logout-button')
    expect(logoutButton.text()).toBe(`${userName} (logout)`)
  })
})