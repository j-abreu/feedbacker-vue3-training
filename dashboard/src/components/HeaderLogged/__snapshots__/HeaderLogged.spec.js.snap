// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLogged /> should render header logged 1`] = `
<div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
  <div class="w-28 lg:w-36"><img class="w-full"></div>
  <div class="flex">
    <ul class="flex list-none">
      <li id="header-credentials-button" class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Credentials </li>
      <li id="header-feedbacks-button" class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Feedbacks </li>
      <li id="logout-button" class="px-6 py-2 font-bold bg-white rounded-full cursor-pointer text-brand-main-focus:outline-none whitespace-nowrap">...</li>
    </ul>
  </div>
</div>
`;
